EIPS/eip-creation_collision.md

```diff
@@ -0,0 +1,74 @@
+---
+eip: <not assigned>
+title: Revert creation in case of collision
+description: Revert contract creation if address already has code
+author: Renan Rodrigues de Souza (@RenanSouza2)
+discussions-to: https://ethereum-magicians.org/t/eip-revert-on-address-collision/13442
+status: Draft
+type: Standards Track
+category: Core
+created: 2023-03-20
+---
+
+## Abstract
+
+There is no current definition to what happens when a contract creation happens in an address with code already deployed. This fix prevents an attack of deploying a contract code and later changing the code arbitrarily.
+
+## Motivation
+
+In EIP-3607 it was estimated that to create an address collision it would take something about 10 billion USD and this number decreases as computers processing power grow.
+
+## Specification
+
+In contract creation where `new contract address` has a `CODEHASH != EMPTYCODEHASH` MUST have its current environment execution reverted, where `EMPTYCODEHASH = 0xc5d2460186f7233c927e7db2dcc703c0e500b653ca82273b7bfad8045d85a470`.
+
+## Rationale
+
+One of the core tenants of smart contracts is it's code being guranteed not to change, with enougth computing power an attacker can change the code stored in an address to any other code allowing them to remove funds or transfer tokens.
```

Comment:
```suggestion
One of the core tenants of smart contracts is its code being guaranteed not to change. With enough computing power an attacker can change the code stored in an address to any other code, allowing them to remove funds or transfer tokens.
```

Reply:
(I was just reading your EIP out of interest and noticed a few tweaks I could suggest.)

Reply:
Thank you